tracing/src/lib.rs

```diff
@@ -174,6 +174,25 @@
 //! # fn main() {}
 //! ```
 //!
+//! For functions which don't have built-in tracing support and can't have
+//! the `#[instrument]` attribute applied (such as from an external crate,
+//! the [`Span` struct][`Span`] has a [`in_scope()` method][`in_scope`]
+//! which can be used to easily wrap synchonous code in a span.
+//!
+//! For example:
+//! ```rust
+//! use tracing::info_span;
+//!
+//! # fn doc() -> Result<(), ()> {
+//! # mod serde_json {
+//! #    pub(crate) fn from_slice(buf: &[u8]) -> Result<(), ()> { Ok(()) }
+//! # }
+//! # let buf = &[];
+//! let json = info_span!("json.parse").in_scope(|| serde_json::from_slice(&buf))?;
```

Comment:
ah, looks like the build is now failing because the type of `buf` is `&[_; 0]`, and when we borrow it, we get `&&[_; 0]`, rather than `&[u8]`. I think changing it to 
```suggestion
//! # let buf: [u8; 0] = [];
//! let json = info_span!("json.parse").in_scope(|| serde_json::from_slice(&buf))?;
```
will fix this while still allowing us to "borrow" `buf` in the example